ImportExport/ObjectsToPython/ObjectsToPython.FCMacro

```diff
@@ -0,0 +1,344 @@
+            
+            if isinstance(obj, Part.Point):
+                return "Part.Point(Vector(%s, %s, %s))"%(floatstr(obj.X), floatstr(obj.Y), floatstr(obj.Z))
+            
+            if isinstance(obj, Part.LineSegment):
+                return "Part.LineSegment(%s, %s)"%(obj.StartPoint, obj.EndPoint)
+            
+            if isinstance(obj, Part.Circle):
+                return "Part.Circle(%s, %s, %s)"%(vecstr(obj.Center), obj.Axis, floatstr(obj.Radius))
+            
+            if isinstance(obj, Part.ArcOfCircle):
+                #return "Part.ArcOfCircle(%s, %s, %s)"%(obj.Center, obj.StartPoint, obj.EndPoint)
+                return "Part.ArcOfCircle(%s, %s, %s)"%(objectToText(obj.Circle), obj.FirstParameter, obj.LastParameter)
+            
+            if isinstance(obj, Part.Ellipse):
+                return "Part.Ellipse(%s, %s, %s)"%(vecstr(obj.Center), obj.MajorRadius, obj.MinorRadius)
+            
+            if isinstance(obj, Part.BSplineCurve):
+                poles = ""
+                komma = False
+                for p in obj.getPoles():
+                    if komma:
+                        poles += ", "
+                    poles += vecstr(p)
+                    komma = True
+                    
+                return "Part.BSplineCurve([%s])"%poles
+            
+            if isinstance(obj, Vector):
+                return vecstr(obj)
+
+            liststart = ""
+            if isinstance(obj, list):
+                liststart = "["
+                listend = "]"
+                
+            if isinstance(obj, tuple):
+                liststart = "("
+                listend = ")"
+                
+            if liststart != "":
+                sline = liststart;
+                comma = False
+                for listele in obj:
+                    if comma:
+                        sline = sline + ", "
+                    else:
+                        comma = True                
+                
+                    val = objectToText(listele, objectlist)
+                    if val is not None:
+                        sline = sline + val
+                    
+                sline += listend;
+                return sline                  
+            
+            stringobj = "%s"%(obj)                            
+            return stringobj
+        
+    
+def addSketch(obj, objname): 
+    prop = obj.getPropertyByName('Geometry')  
+ 
+    for geo in prop:  
+        objstr = objectToText(geo)
+        addScript("%s.addGeometry(%s, %s)"%(objname, objstr, objectToText(geo.Construction)))
+    
+    splinecount = 0
+    concount = 0
+    prop = obj.getPropertyByName('Constraints')
+    for con in obj.Constraints:
+        conargs = ""
+        contype = con.Type
+        if con.Type == 'Coincident':
+            conargs = "%d, %d, %d, %d"%(con.First, con.FirstPos, con.Second, con.SecondPos)
```

Comment:
What about `conargs = '{c.First}, {c.FistPos}, {c.Second}, {c.SecondPos}'.format(c=con)`?

Reply:
good suggestion, but I am too lazy to change my code. Next time I will do it like this